recipes/gdal/all/conanfile.py

```diff
@@ -0,0 +1,650 @@
+        args.append("--with-lerc={}".format("no" if self.options.without_lerc else "yes"))
+        if self.options.with_null:
+            args.append("--with-null")
+        if self.options.get_safe("with_exr") is not None:
+            args.append("--with-exr={}".format("yes" if self.options.with_exr else "no"))
+
+        with tools.chdir(configure_dir):
+            self._autotools.configure(args=args)
+        return self._autotools
+
+    def _unix_path(self, path):
+        if self.settings.os == "Windows":
+            return tools.unix_path(path)
+        return path
+
+    def build(self):
+        self._patch_sources()
+        if self.settings.compiler == "Visual Studio":
+            with tools.chdir(self._source_subfolder):
+                with tools.vcvars(self.settings):
+                    with tools.environment_append(VisualStudioBuildEnvironment(self).vars):
+                        self.run("nmake -f makefile.vc {}".format(" ".join(self._get_nmake_args())))
+        else:
+            autotools = self._configure_autotools()
+            with tools.chdir(self._source_subfolder):
+                autotools.make()
+
+    def package(self):
+        self.copy("LICENSE.TXT", dst="licenses", src=self._source_subfolder)
+        if self.settings.compiler == "Visual Studio":
+            with tools.chdir(self._source_subfolder):
+                with tools.vcvars(self.settings):
+                    with tools.environment_append(VisualStudioBuildEnvironment(self).vars):
+                        self.run("nmake -f makefile.vc devinstall {}".format(" ".join(self._get_nmake_args())))
+            tools.rmdir(os.path.join(self.package_folder, "data"))
+            for pdb_file in glob.glob(os.path.join(self.package_folder, "lib", "*.pdb")):
+                os.remove(pdb_file)
+        else:
+            autotools = self._configure_autotools()
+            with tools.chdir(self._source_subfolder):
+                autotools.install()
+            tools.rmdir(os.path.join(self.package_folder, "share"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "gdalplugins"))
+            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+            for la_file in glob.glob(os.path.join(self.package_folder, "lib", "*.la")):
+                os.remove(la_file)
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "GDAL"
+        self.cpp_info.names["cmake_find_package_multi"] = "GDAL"
+        self.cpp_info.names["pkg_config"] = "gdal"
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs.extend(["dl", "m"])
+            if self.options.threadsafe:
+                self.cpp_info.system_libs.append("pthread")
+        elif self.settings.os == "Windows":
+            self.cpp_info.system_libs.extend(["psapi", "ws2_32"])
```

Comment:
GDAL directly call functions defined in `kernel32` if using win32 threads. Should I add this lib also? It's not required by MinGW Posix Threads I think.

Reply:
No, kernel32 is not required. I used to write code with WinAPI in past, including windows threads, kernel32 was not required.

Reply:
functions like [CreateThread](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createthread) reside in **kernel32.lib**. it's usually included by default in Visual Studio, but for other compilers (MinGW, Clang, Intel, Borland, etc.) it might not be the case.